R-package/R/optimizer.R

```diff
@@ -36,7 +36,7 @@ mx.opt.sgd <- function(learning.rate,
         sgd[[indexKey]] <- 0
       } else {
         indexValue <- sgd[[indexKey]]
-        sgd[[indexKey]] <- indexValue
+        sgd[[indexKey]] <- indexValue + 1
```

Comment:
This is to update the `num.update`. I am afraid the `num.update` won't be changed if we don't increase `sgd[[indexKey]]`.

The same thing has been done in https://github.com/dmlc/mxnet/blob/master/python/mxnet/optimizer.py#L256-L257